### PR TITLE
addition of all claimBuyerFund-tests

### PR DESCRIPTION
**Pull request by josojo**

### CODE REVIEW

File: test/dutchExchange-claimBuyerFunds.js [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+    await postBuyOrder(eth, gno, auctionIndex, 10e18, buyer2)
+    await waitUntilPriceIsXPercentOfPreviousPrice(eth, gno, 0.6)
+    await postBuyOrder(eth, gno, auctionIndex, 10e18, buyer2)
+
+
+    // Check extra Token balance 
+    const [claimedAmount] = (await dx.claimBuyerFunds.call(eth.address, gno.address, buyer1, auctionIndex)).map(i => i.toNumber())
+    const [num, den] = (await dx.closingPrices.call(eth.address, gno.address, auctionIndex))
+    await dx.claimBuyerFunds(eth.address, gno.address, buyer1, auctionIndex)
+    assert.equal(((bn(valMinusFee(10e18)).div(num).mul(den)).add(extraTokensAvailable.div(bn(2))))
```
Comment (Velenir) [inline test/dutchExchange-claimBuyerFunds.js]: Unnecessary `bn(2)`, can just be `2`

File: test/dutchExchange-claimBuyerFunds.js [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+    const extraTokensAvailable = await dx.extraTokens.call(eth.address, gno.address, 2)
+    await postBuyOrder(eth, gno, auctionIndex, 10e18, buyer1)
+    await postBuyOrder(eth, gno, auctionIndex, 10e18, buyer2)
+    await waitUntilPriceIsXPercentOfPreviousPrice(eth, gno, 0.6)
+    await postBuyOrder(eth, gno, auctionIndex, 10e18, buyer2)
+    const balanceOfBuyer1 = await dx.balances.call(eth.address, buyer1)
+    const [claimedAmount] = (await dx.claimBuyerFunds.call(eth.address, gno.address, buyer1, auctionIndex)).map(i => i.toNumber())
+    const [num, den] = (await dx.closingPrices.call(eth.address, gno.address, auctionIndex))
+    await dx.claimBuyerFunds(eth.address, gno.address, buyer1, auctionIndex)
+    assert.equal(((bn(valMinusFee(10e18)).div(num).mul(den)).add(extraTokensAvailable.div(bn(2))))
```
Comment (Velenir) [inline test/dutchExchange-claimBuyerFunds.js]: Unnecessary `bn(2)`, can just be `2`

File: test/dutchExchange-claimBuyerFunds.js [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+    // checking that closingPriceToken.num == 0
+    const [closingPriceNumToken] = (await dx.closingPrices.call(eth.address, gno.address, auctionIndex)).map(i => i.toNumber())
+    assert.equal(closingPriceNumToken, 0)
+    
+    // actual testing at time with previous price
+    const [claimedAmount] = (await dx.claimBuyerFunds.call(gno.address, eth.address, buyer2, auctionIndex)).map(i => i.toNumber())
+    const [num, den] = (await dx.getPriceForJS.call(gno.address, eth.address, auctionIndex)).map(i => i.toNumber())
+    let sellVolume = (await dx.sellVolumesCurrent.call(gno.address, eth.address))
+    let buyVolume = await dx.buyVolumes.call(gno.address, eth.address)
+    let oustandingVolume = (sellVolume.mul(bn(num)).div(den)).sub(buyVolume)
```
Comment (Velenir) [inline test/dutchExchange-claimBuyerFunds.js]: You don't need `bn(num)` here, what you pas to `BN.method()` is automatically wrapped in `bn`

File: test/dutchExchange-claimBuyerFunds.js [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+
+    await waitUntilPriceIsXPercentOfPreviousPrice(eth, gno, 0.4)
+
+    // checking that closingPriceToken.num == 0
+    const [closingPriceNumToken] = (await dx.closingPrices(eth.address, gno.address, auctionIndex)).map(i => i.toNumber())
+    assert.equal(closingPriceNumToken, 0)
+    
+    // actual testing
+    const [claimedAmount] = (await dx.claimBuyerFunds.call(eth.address, gno.address, buyer1, auctionIndex)).map(i => i.toNumber())
+    assert.equal(bn(valMinusFee(totalSellAmount2ndAuction)), claimedAmount)
```
Comment (Velenir) [inline test/dutchExchange-claimBuyerFunds.js]: Shouldn't wrap `valMinusFee(totalSellAmount2ndAuction)` in `bn()`. Assert doesn't fail here thanks to loose equality conversion, but it won't work in all cases.

**Comment by josojo** [comment]
closing since it is in new PR #178

**Pull request closed without merging**